table/block.cc

```diff
@@ -170,6 +170,23 @@ void DataBlockIter::Seek(const Slice& target) {
     return;
   }
   uint32_t index = 0;
+
+  // we use the hash index
+  // if 1) the caller is doing point lookup
+  //   i.e. data_block_hash_index is not nullptr
+  // and 2) data_block_hash_index is valid
+  //   i.e. the block content contains hash map
+  if (data_block_hash_index_) {
```

Comment:
Instead of complicating logic in DataBlockIter::Seek(), can you create a new function like DataBlockIter::SeekForGet() or something like that?